(POC) run_container plan function (#2728)
puppetlabs/bolt: documentation/experimental_features.md

Bolt team attempts to make these changes painless by providing useful warnings
around breaking behavior where possible. 

## `run_container` plan step

> This feature was introduced in TODO.

Reviewer: I'm guessing this will go out next release?

> Currently available as a Puppet plan function, `run_container()` will run a provided image and

Reviewer: ```suggestion
Currently available as a Puppet plan function, `run_container()` runs a provided image and
```

> return the `stdout`, `stderr`, and `exit_code` for that image. The function supports several

Reviewer: ```suggestion
returns the `stdout`, `stderr`, and `exit_code` for that image. The function supports several
```

options, including `cmd` to specify a command to run in the container, `volumes` to mount volumes to
the container, and `ports` to publish ports from the container on the host. You can see all the
supported options in the [plan function documentation](plan_functions.md#run_container).

The function runs the container on the Bolt controller, not on remote targets. This function is
supported on both \*nix and Windows systems. 

This plan clones the [Relay repository](https://github.com/puppetlabs/relay), builds the Go binary in a

> container that has all the dependencies in it, then installs it to a local path.

Reviewer: ```suggestion
container that has all the dependencies in it, and installs the binary to a local path.
```
